src/docker.ts

```diff
@@ -26,13 +32,30 @@ const saveDockerImages = async (): Promise<void> => {
       `Cache miss occurred on the primary key ${key}. Not saving cache as read-only option was selected.`
     );
   } else {
-    await execBashCommand(
-      'docker image list --format "{{ .Repository }}:{{ .Tag }}" | ' +
-        '2>&1 xargs --delimiter="\n" --no-run-if-empty --verbose --exit ' +
-        `docker save --output ${DOCKER_IMAGES_PATH}`
+    const preexistingImages = getState(DOCKER_IMAGES_LIST).split("\n");
+    const images = await execBashCommand(
+      'docker image list --format "{{ .Repository }}:{{ .Tag }}"'
     );
-    await saveCache([DOCKER_IMAGES_PATH], key);
+    const imagesList = images.split("\n");
+    const newImages = imagesList.filter(
+      (image) => !preexistingImages.includes(image)
```

Comment:
I'm not confident about the syntax.  I think this might be what you mean, but can you confirm?!
```
const newImages: string[] = imagesList.filter(
      (image: string) => !preexistingImages.includes(image)
    );
```

Reply:
TypeScript should be able to infer the type of `newImages`. I meant to specify the return type of the callback function like you have elsewhere, so you should only need to modify line 41.

Reply:
There is no need to specify the type of `newImages`.